types/validation.go

```diff
@@ -427,3 +468,113 @@ func verifyBasicValsAndCommit(vals *ValidatorSet, commit *Commit, height int64,
 
 	return nil
 }
+
+func verifyAggregatedCommit(
+	chainID string,
+	vals *ValidatorSet,
+	commit *Commit,
+	votingPowerNeeded int64,
+	ignoreSig func(CommitSig) bool,
+	countSig func(CommitSig) bool,
+	lookUpByIndex bool,
+) error {
+	var (
+		val                *Validator
+		valIdx             int32
+		seenVals           = make(map[int32]int, len(commit.Signatures))
+		talliedVotingPower int64
+		aggSig1, aggSig2   []byte
+		msg1, msg2         []byte
+	)
+
+	pubkeys1 := make([]*bls12381.PubKey, 0, len(commit.Signatures))
+	pubkeys2 := make([]*bls12381.PubKey, 0, len(commit.Signatures))
+
+	for idx, commitSig := range commit.Signatures {
```

Comment:
As discussed, it'd be great if we can explicitly check that:
* there is _exactly_ 1 occurrence of `BlockIDFlagAggCommit`
* there is _at most_ 1 occurrence of `BlockIDFlagAggNil`
* there are 0 occurrences or `BlockIDFlagCommit`
* there are 0 occurrences or `BlockIDFlagNil`